spinn_front_end_common/interface/interface_functions/router_provenance_gatherer.py

```diff
@@ -135,288 +180,236 @@ def _add_router_provenance_data(
-                        x, y,
-                        reinjection_status.n_dropped_packet_overflows))))
-            items.append(ProvenanceDataItem(
-                self.__add_name(names, "Reinjected"),
-                reinjection_status.n_reinjected_packets))
-            items.append(ProvenanceDataItem(
-                self.__add_name(names, "Dumped_from_a_Link"),
-                str(reinjection_status.n_link_dumps),
-                report=(reinjection_status.n_link_dumps > 0 and
-                        self._has_virtual_chip_connected(self._machine, x, y)),
-                message=(
-                    "The extra monitor on {}, {} has detected that {} packets "
-                    "were dumped from a outgoing link of this chip's router."
-                    " This often occurs when external devices are used in the "
-                    "script but not connected to the communication fabric "
-                    "correctly. These packets may have been reinjected "
-                    "multiple times and so this number may be a overestimate."
-                    .format(x, y, reinjection_status.n_link_dumps))))
-            items.append(ProvenanceDataItem(
-                self.__add_name(names, "Dumped_from_a_processor"),
-                str(reinjection_status.n_processor_dumps),
-                report=reinjection_status.n_processor_dumps > 0,
-                message=(
-                    "The extra monitor on {}, {} has detected that {} packets "
-                    "were dumped from a core failing to take the packet."
-                    " This often occurs when the executable has crashed or"
-                    " has not been given a multicast packet callback. It can"
-                    " also result from the core taking too long to process"
-                    " each packet. These packets were reinjected and so this"
-                    " number is likely a overestimate.".format(
-                        x, y, reinjection_status.n_processor_dumps))))
-
-        items.append(ProvenanceDataItem(
-            self.__add_name(names, "Error status"),
-            str(router_diagnostic.error_status),
-            report=router_diagnostic.error_status > 0,
-            message=(
-                "The router on {}, {} has a non-zero error status.  This could"
-                " indicate a hardware fault.  The errors set are {}, and the"
-                " error count is {}".format(
-                    x, y, router_diagnostic.errors_set,
-                    router_diagnostic.error_count))))
-
-    @staticmethod
-    def _has_virtual_chip_connected(machine, x, y):
+            message=self.__DROPPED_MC_MSG.format(
+                x, y, diagnostics.n_dropped_multicast_packets))
+        yield ProvenanceDataItem(
+            names + ["Dropped_Multicast_Packets_via_local_transmission"],
+            diagnostics.user_3,
+            report=(diagnostics.user_3 > 0),
+            message=self.__DROPPED_LOCAL_MC_MSG.format(
```

Comment:
I know this method is very long but moving the Strings out only makes it even harder to read!

Reply:
Fix for one of these is fix for all.